### ISSUE TITLE
Incorrect error column position for lines containing non-ASCII characters

### ISSUE THREAD

**Issue by ZhenShuo2021** [rvben/rumdl#670]
### rumdl version

0.2.10

### Installation method

Pre-built binary

### Bug description

Error column positions are reported incorrectly when the source text contains non-ASCII characters.

### Steps to reproduce

Create a Markdown file with the following content:

```md
foo foo [foo](not-exist.md) bar
你好你好[你好](not-exist.md) bar
```

The error highlight in first line is `not-exist.md`, but the second line highlights `bar`.

### Expected behavior

The error highlight points to `not-exist.md` on both lines.

### Actual behavior

The error highlight correctly points to `not-exist.md` on the first line, but incorrectly points to `bar` on the second line.

### Configuration

```toml

```

### Operating system

macOS

**Comment by rvben** [comment]
Fixed in **v0.2.18**. Error/warning columns are now character-based instead of byte-based, so lines containing non-ASCII characters report the correct column. Verified on your repro (`你好你好[你好](not-exist.md)` now reports column 10). Thanks for the clear report, @ZhenShuo2021!
